### PR TITLE
plugin: re-organize `proposed` api files

### PR DESCRIPTION
**Pull request by vince-fugnitto**
#### What it does

The pull-request re-organizes our `theia-proposed.d.ts` to follow the same conventions present in vscode, and also make it easier to update or add proposed apis in the future.

The changes:
- split the proposed apis by sections into their own files
- add the "extra" apis to `theia-extra.d.ts` properly

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Review by alvsan09 — APPROVED** [review]
I have reviewed the distribution of code to the different new files, 
I found all code previously hosted in the former `theia-proposed.d.ts` file
mapped to the new files. :+1: 

The additions and modifications look good to me !

Testing was limited to start `theia` validate that no new activation failures arise,
and test some basic functionality.

**Comment by JonasHelming** [comment]
@rschnekenbu FYI

File: packages/plugin/src/theia.proposed.textSearchQuery.d.ts
Comment (rschnekenbu) [inline packages/plugin/src/theia.proposed.textSearchQuery.d.ts:1]: Shall this file be named theia.proposed.textSearchProvider.ts to match naming from vscode? That would facilitate the updates later and also tooling to compare API coverage

File: packages/plugin/src/theia.proposed.textSearchQuery.d.ts [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+        /**
+         * Whether or not to search for whole word matches only.
+         */
+        isWordMatch?: boolean;
+    }
+
+    /**
+     * Options that can be set on a findTextInFiles search.
+     */
+    export interface FindTextInFilesOptions {
```
Comment (rschnekenbu) [inline packages/plugin/src/theia.proposed.textSearchQuery.d.ts:58]: This is a duplicate from theia.proposed.findTextInFiles.d.ts. This one may be removed?

**Comment by vince-fugnitto** [comment]
@rschnekenbu thank you for the feedback, I updated the pull-request accordingly.

**Review by rschnekenbu — APPROVED** [review]
@vince-fugnitto, that looks good to me!

**Pull request merged**